monkey/monkey_island/cc/next_ui/src/_components/app-bar/AppBar.tsx

```diff
@@ -0,0 +1,117 @@
+        setAnchorElUser(null);
+    };
+
+    const handleRouteClick = (path: string) => {
+        router.push(path);
+    };
+
+    return (
+        <AppBar
+            id={classes['app-bar']}
+            position="static"
+            color="primary"
+            enableColorOnDark>
+            <Container maxWidth="xl">
+                <Toolbar disableGutters>
+                    <Box
+                        className={'app-logo-container'}
+                        onClick={() =>
+                            handleRouteClick(AUTHENTICATED_PATHS.ROOT)
+                        }>
+                        <SvgIcon inheritViewBox={true}>
+                            <MonkeyHeadSvg />
+                        </SvgIcon>
+                        <SvgIcon inheritViewBox={true}>
+                            <MonkeyNameSvg />
+                        </SvgIcon>
+                    </Box>
+                    <Box className={'app-links'}>
+                        {ALL_LINKS.map((currPath) => (
+                            <div
+                                key={currPath.label}
+                                className={'app-route-link'}
+                                onClick={() => handleRouteClick(currPath.path)}>
+                                <Typography>{currPath.label}</Typography>
+                            </div>
+                        ))}
+                    </Box>
+                    <Box className={'profile-wrapper'}>
+                        <Tooltip title="Open settings">
+                            <IconButton
+                                onClick={handleOpenUserMenu}
+                                sx={{ p: 0 }}>
+                                <Avatar
+                                    alt="Remy Sharp"
+                                    src="/static/images/avatar/2.jpg"
+                                />
+                            </IconButton>
+                        </Tooltip>
+                        <Menu
+                            sx={{ mt: '45px' }}
+                            id="menu-appbar"
+                            anchorEl={anchorElUser}
+                            anchorOrigin={{
+                                vertical: 'top',
+                                horizontal: 'right'
+                            }}
+                            keepMounted
+                            transformOrigin={{
+                                vertical: 'top',
+                                horizontal: 'right'
+                            }}
+                            open={Boolean(anchorElUser)}
+                            onClose={handleCloseUserMenu}>
+                            <MenuItem onClick={handleCloseUserMenu}>
+                                <LogoutButton />
+                            </MenuItem>
+                        </Menu>
+                    </Box>
+                </Toolbar>
+            </Container>
+        </AppBar>
+    );
+};
```

Comment:
This component is very large. Consider extracting the subcomponents into function components.